cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/SampleTypeFieldChecker.ts

```diff
@@ -0,0 +1,209 @@
+      type: 'string',
+    },
+  ],
+};
+// tslint:disable-next-line: no-console
+// tslint:disable-next-line: no-console
+console.log(recordField.fields);
+
+const complexArrField2: IArrayFieldNullable = {
+  name: 'arr1',
+  type: [
+    {
+      type: 'array',
+      items: 'string',
+    },
+    'null',
+  ],
+};
+// tslint:disable-next-line: no-console
+console.log(
+  Array.isArray(complexArrField2.type)
+    ? complexArrField2.type.find((t) => t !== 'null' && typeof t.items !== 'undefined')
+    : complexArrField2.type
+);
+
+const isNullable = (type) => Array.isArray(type) && type.find((t) => t === 'null');
+const complexArrayField: IArrayFieldNullable = {
+  name: 'complexArray',
+  type: [
+    {
+      type: 'array',
+      items: [
+        {
+          type: 'record',
+          name: 'ad5bddf76ef2743218d79d3905f0f8e4f',
+          fields: [
+            {
+              name: 'name',
+              type: 'string',
+            },
+            {
+              name: 'email',
+              type: 'string',
+            },
+          ],
+        },
+        'null',
+      ],
+    },
+    'null',
+  ],
+};
+
+if (Array.isArray(complexArrayField.type)) {
+  const a1 = complexArrayField.type.find((t) => t !== 'null' && t.type === 'array');
+  // tslint:disable-next-line: no-console
+  console.log(a1 !== 'null' && a1.items);
+  if (isNullable(complexArrayField.type)) {
+    // tslint:disable-next-line: no-console
+    console.log('nullable is true');
+  }
+}
+
+const complexUnionField: IUnionField = {
+  name: 'something',
+  type: [
+    'long',
+    {
+      type: 'map',
+      keys: {
+        type: 'record',
+        name: 'a64d56b7343854e81801874b77b536802',
+        fields: [
+          {
+            name: 'sdfsd',
+            type: 'string',
+          },
+          {
+            name: 'sdfsdsdfsdf',
+            type: 'string',
+          },
+        ],
+      },
+      values: 'string',
+    },
+    {
+      type: 'record',
+      name: 'record1',
+      fields: [
+        {
+          name: 'name',
+          type: 'string',
+        },
+        {
+          name: 'email',
+          type: 'string',
+        },
+      ],
+    },
+  ],
+};
+const isSimpleType = (type: ISimpleType | IComplexTypeFieldNullable) =>
+  typeof type === 'string' &&
+  [
+    'boolean',
+    'bytes',
+    'date',
+    'decimal',
+    'double',
+    'float',
+    'int',
+    'long',
+    'number',
+    'string',
+    'time',
+  ].indexOf(type) !== -1;
+if (Array.isArray(complexUnionField.type)) {
+  const map1 = complexUnionField.type
+    .filter((t) => typeof t !== 'string')
+    .find((t: IComplexType) => !Array.isArray(t.type) && t.type === 'map') as IMapFieldBase;
+  let fieldsInRecords;
+  if (!Array.isArray(map1.keys) && typeof map1.keys === 'object' && map1.keys.type === 'record') {
+    fieldsInRecords = (map1.keys as IRecordField).fields;
+  }
+  // tslint:disable-next-line: no-console
+  console.log(map1, fieldsInRecords);
```

Comment:
nit: are these console.log necessary?

Reply:
This file is not being used anywhere.. This is to playaround with schema and typescript types. We can safely ignore this file.